### ISSUE TITLE
Blinds with 3rd position hex command

### ISSUE THREAD

**Issue by afbenevides** [lprhodes/homebridge-broadlink-rm#287]
Hi, first thanks for this great plugin, it is very useful. I would like to help in creating a new Blind accessory type for this plugin that has 4 options/buttons: Goto Open position, Goto Close position, Stop, Goto 3rd position. (4 different hex codes). I'm ok in python and never done javascript before, but I'm willing to learn in the way.  The Goto open Button will set the blind to open, and with this kind of motor that I have (Dooya DM25LEQ) it stops automatically when arriving at the configured open position same thing for close and 3rd position. You can configure 3 positions on the motor: Open, Close and 3rd position ( anything in between). 

So it seems not too complicated for a project for me if I can base my work on the existing blind accessory that has open, close, stop function, but to make it more work like 4 momentary switches embedded in one accessory ( like my remote). 4 buttons, 4 hex codes, all in the same accessory. If on top of that we can still keep the animation of the blinds it will be perfect. (with still a parameter of open delay and close delay, with a % equaling the middle 3rd position)

For now i'm using the existing plugin with 4 different switches for each blind. Would be much cleaner if I could regroup in one accessory.

I will begin reading the files for switch and blinds to see if I can make that happen even without javascript experience..... If you have any advice on where to begin, and on the documentation I could use please tell me.

**Comment by lprhodes** [comment]
Hey @afbenevides 

How come you can't use the current window-covering accessory as it is in order to put the blinds to specific percentages?

**Comment by afbenevides** [comment]
Hi @lprhodes 

In fact I could, but in this config if I understood well the position is controlled by assigning a travel time equal to the percentage of travel to go on total travel. On my ZEBRA blinds, we need the 3rd position to be precisely positioned so the stripes get really aligned. (transparent stripes aligned and white stripes aligned). So for that the motor already has a third limit position that can be reached if the hex code for this position is sent. So 4 hex codes corresponding to open, close, stop and this third position. Your idea makes me think that for all other positions, it would be nice to still let % control.

**Comment by lprhodes** [comment]
Interesting, I'd love to see a video once it's complete.

By setting the total travel time you should then get 100 different points at which the blind could be stopped which should surely be enough to get things lined up?

**Comment by lprhodes** [comment]
Also, I have some significant amends that I'll be releasing in the next couple of hours so I wouldn't suggest  changing any code right now.

**Comment by lprhodes** [comment]
In the next update every single hex code will be configurable to allow for repeats, multiple hex codes and delays. Which means I can get rid of the extra switch accessories and all the other accessories will be more flexible

**Comment by lprhodes** [comment]
The update is out.

I definitely think you'll be able to use the current accessory to achieve what you want

**Comment by afbenevides** [comment]
HI @lprhodes, I want to try a modification I made on you window covering accessory to include my third hex code. When I try to launch home bridge with this command I found on home bridge readme file:

DEBUG=* homebridge -D -P homebridge-broadlink-rm/

in wich homebridge-broadlink-rm/ is the modified version, I get This Error:

====================
[3/15/2018, 5:58:57 AM] ERROR LOADING PLUGIN homebridge-broadlink-rm:
[3/15/2018, 5:58:57 AM] Error: Cannot find module 'homebridge-platform-helper'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/pi/homebridge-broadlink-rm/platform.js:1:94)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
[3/15/2018, 5:58:57 AM] ====================

So I tried to put this module homebridge-platform-helper somewhere inside the broadlink module, but still did not work.

Can you help me so it finds homebridge-platform-helper. Or redirect me to a good ressource that explains how to load and debug homebridge plugins directly.

Thanks,

**Comment by lprhodes** [comment]
I just do the following:

```
cd ~/Dev/homebridge-broadlink-rm/
homebridge -P ./
```

**Comment by lprhodes** [comment]
My suggestion would be to get it running locally on your machine for quick testing rather than using the rPi

**Comment by afbenevides** [comment]
Thanks I tried it with your command, but still not working.
it seems to be this line that does not work.
const { HomebridgePlatform } = require('homebridge-platform-helper');
 Do you know how can I be sure that the module is there or installed? Or where si I have to put it?

Thanks again for your help, I'm new with all this thing...

**Comment by lprhodes** [comment]
when you clone this repo you then need to run `npm install`

**Comment by afbenevides** [comment]
OK thanks it worked now.

So the code worked. What I did is this modification:

in:
async setTargetPositionActual (hexData, previousValue) {

changed:
const { open, close, stop, third_hex, third_position, third_span } = data;

and  hex sending logic:

if (state.targetPosition > (third_position - third_span) && state.targetPosition < (third_position + third_span)){
      hexData = third_hex
    }
    else {
      hexData = state.opening ? open : close
    }

For me third position is somewhere near 5% so for easier access I created a band from (span of 3) 2 to 8 that send my third hex key... 

It is a work around, since If I understand well we do not have access to modify the actual look of the window covering homekit tile popup. I would have liked to add a dedicated button or switch beside the scroll bar for this 3rd limit....  But still, it works

I will have to correct logic since on my motors, full up, full down and third limit do not need to send a stop hex. I will keep the timing feature to have all the other positions possibilities.

But for now thanks a lot, I have officially begin to do custom things with homebridge with your help.
